code/modules/admin/topic.dm

```diff
@@ -2289,6 +2289,206 @@
+		var/reply_to = locate(href_list["replyto"])
+		var/destination
+		var/notify
+
+		var/obj/item/paper/P = new /obj/item/paper(null) //hopefully the null loc won't cause trouble for us
+
+		if(!fax)
+			var/list/departmentoptions = GLOB.alldepartments + "All Departments"
+			destination = input(usr, "To which department?", "Choose a department", "") as null|anything in departmentoptions
+			if(!destination)
+				qdel(P)
+				return
+
+			for(var/thing in GLOB.allfaxes)
+				var/obj/machinery/photocopier/faxmachine/F = thing
+				if(destination != "All Departments" && F.department == destination)
+					fax = F
+
+
+		var/input_text = input(src.owner, "Please enter a message to send a fax via secure connection. Use <br> for line breaks. Both pencode and HTML work.", "Outgoing message from Centcomm", "") as message|null
+		if(!input_text)
+			qdel(P)
+			return
+
+		var/obj/item/pen/admin_writer = new /obj/item/pen(null)
+
+		input_text = P.parsepencode(input_text, admin_writer, usr) // Encode everything from pencode to html
+		qdel(admin_writer)
+
+		var/customname = input(src.owner, "Pick a title for the fax.", "Fax Title") as text|null
+		if(!customname)
+			customname = "paper"
+
+		var/sendername
+		switch(faxtype)
+			if("Central Command")
+				sendername = "Central Command"
+			if("Syndicate")
+				sendername = "UNKNOWN"
+			if("Custom")
+				sendername = input(owner, "What organization does the fax come from? This determines the prefix of the paper (i.e. Central Command- Title). This is optional.", "Organization") as text|null
+
+		if(sender)
+			notify = alert(owner, "Would you like to inform the original sender that a fax has arrived?","Notify Sender","Yes","No")
+
+		// Create the reply message
+		if(sendername)
+			P.name = "[sendername]- [customname]"
+		else
+			P.name = "[customname]"
+		P.info = input_text
+		P.update_icon()
+		P.x = rand(-2, 0)
+		P.y = rand(-1, 2)
+
+		if(destination != "All Departments")
+			if(fax.receivefax(P) == FALSE)
+				to_chat(owner, "<span class='warning'>Message transmission failed.</span>")
+				return
+		else
+			for(var/thing in GLOB.allfaxes)
+				var/obj/machinery/photocopier/faxmachine/F = thing
+				if(F.z in GLOB.station_z_levels)
+					addtimer(CALLBACK(src, .proc/handle_sendall, F, P), 0)
+
+		var/datum/fax/admin/A = new /datum/fax/admin()
+		A.name = P.name
+		A.from_department = faxtype
+		if(destination != "All Departments")
+			A.to_department = fax.department
+		else
+			A.to_department = "All Departments"
+		A.origin = "Custom"
+		A.message = P
+		A.reply_to = reply_to
+		A.sent_by = usr
+		A.sent_at = world.time
+
+		to_chat(src.owner, "<span class='notice'>Message transmitted successfully.</span>")
+		if(notify == "Yes")
+			var/mob/living/carbon/human/H = sender
+			if(istype(H) && H.stat == CONSCIOUS && (istype(H.ears, /obj/item/device/radio/headset)))
+				to_chat(sender, "Your headset pings, notifying you that a reply to your fax has arrived.")

```

Comment:
All chat sent to Goonchat needs to be marked up with span tags, this'll be really relevant in the future if we try to "dark mode" this window too.